aten/src/ATen/native/mps/operations/View.mm

```diff
@@ -666,69 +38,9 @@ static IntArrayRef updateTensorBaseShape(const Tensor& self) {
-//                T = Tensor
-//                 ----------
-//                 | Orig T |
-//                 ----------
-//                /     |     \
-//             View T  View T  NonView T
-//             /      /    \      |
-//            View T /      \     |
-//            |     /        \    |
-//            |    /          \   |
-//            |   /            \  |
-//            NonView T         NonView T
-static ViewCachedGraph* createViewGraph(const Tensor& self,
-                                        const Tensor& updates,
-                                        IntArrayRef size,
-                                        IntArrayRef stride,
-                                        int64_t storage_offset,
-                                        bool needsScatter) {
-  IntArrayRef base_shape = updateTensorBaseShape(self);
-
-  @autoreleasepool {
-    string key = getStridedKey(
-        self.scalar_type(), updates.scalar_type(), base_shape, size, stride, storage_offset, needsScatter);
-    return LookUpOrCreateCachedGraph<ViewCachedGraph>(key, [&](auto mpsGraph, auto newCachedGraph) {
-      MPSGraphTensor* updatesTensor = nil;
-      // Workaround for MPSShaderLibrary bug in macOS Monterey
-      // This is fixed in macOS Ventura
-      auto inputType = getMPSScalarType(self.scalar_type());
-      if (inputType == MPSDataTypeUInt8) {
-        inputType = MPSDataTypeInt8;
-      }
-
-      // Self is the input tensor we are creating view of
-      newCachedGraph->inputTensor = mpsGraphRankedPlaceHolder(mpsGraph, inputType, getMPSShape(base_shape));
-      newCachedGraph->storageOffsetTensor = mpsGraphRankedPlaceHolder(mpsGraph, MPSDataTypeInt32, @[ @1 ]);
-      for ([[maybe_unused]] const auto i : c10::irange(size.size())) {
-        newCachedGraph->strideTensors.push_back(mpsGraphRankedPlaceHolder(mpsGraph, MPSDataTypeInt32, @[ @1 ]));
-      }
-      if (needsScatter) {
-        auto updatesType = getMPSScalarType(updates.scalar_type());
-        if (updatesType == MPSDataTypeUInt8) {
-          updatesType = MPSDataTypeInt8;
-        }
-        newCachedGraph->updatesTensor = mpsGraphRankedPlaceHolder(mpsGraph, updatesType, getMPSShape(self.numel()));
-        updatesTensor = newCachedGraph->updatesTensor;
-        if (inputType != updatesType) {
-          updatesTensor = [mpsGraph castTensor:updatesTensor toType:inputType name:@"castUpdatesTensor"];
-        }
-      }
-      newCachedGraph->outputTensor =
-          chainViewOperation(newCachedGraph, size, stride, storage_offset, base_shape, needsScatter, updatesTensor);
-    });
-  }
-}
-
 static std::string getGatherScatterFunctionName(ScalarType scalarType, int64_t dim, bool needsScatter) {
   std::string kernelName = needsScatter ? "scatter" : "gather";
-  return kernelName + "_kernel_" + std::to_string(dim == 0 ? 1 : dim);
+  return kernelName + "_kernel_" + (dim < 5 ? std::to_string(dim == 0 ? 1 : dim) : "n");
```

Comment:
Is the idea here that we have special case for 1-4 dimensions and then a shared kernel for 5+?
The reason we do that is that the 1-4 have explicit args which is faster to pass in while the 5+ is pointer-based passed and so more expensive?
This would be worth a small comment maybe?

Also what's the max number of dimensions accepted here? Do we test for it?

Reply:
Let me leave a comment. 
I don't think passing args using packed_int makes kernel faster, but rather explicit loop unrolling and lack of boundary checks. I.e. even on cpu `int sum = 0; for(int i = 0; i < n; ++i) sum = v[i];` is 3x slower than `sum = v[0] + v[1];` if one can infer at compile time that `n` is 2